extensions/standard-processors/tests/unit/ListenUDPTests.cpp

```diff
@@ -29,49 +29,67 @@ using namespace std::literals::chrono_literals;
 
 namespace org::apache::nifi::minifi::test {
 
-constexpr uint64_t PORT = 10256;
+void check_no_error(std::error_code error_code) {
+  CHECK_FALSE(error_code);
+}
 
-void check_for_attributes(core::FlowFile& flow_file) {
+void check_for_attributes(core::FlowFile& flow_file, uint16_t port) {
   const auto local_addresses = {"127.0.0.1", "::ffff:127.0.0.1", "::1"};
-  CHECK(std::to_string(PORT) == flow_file.getAttribute("udp.port"));
+  CHECK(std::to_string(port) == flow_file.getAttribute("udp.port"));
   CHECK(ranges::contains(local_addresses, flow_file.getAttribute("udp.sender")));
 }
 
+uint16_t scheduleProcessorOnRandomPort(SingleProcessorTestController& controller, const std::shared_ptr<ListenUDP>& listen_udp) {
```

Comment:
there seem to be three implementations of this functions, differing in which kind or processor they schedule, could a single template function achieve the same?